onebusaway-gtfs-transformer/src/main/java/org/onebusaway/gtfs_transformer/impl/AnomalyCheckFutureTripCounts.java

```diff
@@ -0,0 +1,314 @@
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.YEAR, date.getYear());
+        calendar.set(Calendar.MONTH, date.getMonth()-1);
+        calendar.set(Calendar.DATE, date.getDay());
+        Date date1 = calendar.getTime();
+        date1 = removeTime(date1);
+        return date1;
+    }
+
+    private Date removeTime(Date date) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        calendar.set(Calendar.HOUR_OF_DAY, 0);
+        calendar.set(Calendar.MINUTE, 0);
+        calendar.set(Calendar.SECOND, 0);
+        calendar.set(Calendar.MILLISECOND, 0);
+        date = calendar.getTime();
+        return date;
+    }
+
+    private String getTopic() {
+        return System.getProperty("sns.topic");
+    }
+
+    public void setDatesToIgnoreUrl(String url){
+        this.datesToIgnoreUrl = url;
+    }
+
+    public void setDatesToIgnoreFile(String url){
+        this.datesToIgnoreFile = url;
+    }
+
+    public void setDayAvgTripMapUrl(String url){
+        this.dayAvgTripMapUrl = url;
+    }
+
+    public void setDayAvgTripMapFile(String url){
+        this.dayAvgTripMapFile = url;
+    }
+
+    public void setHolidaysUrl(String url){
+        this.holidaysUrl = url;
+    }
+
+    public void setHolidaysFile(String url){
+        this.holidaysFile = url;
+    }
+
+    public void setPercentageMatch(double percentageMatch) {
+        this.percentageMatch = percentageMatch;
+    }
+
+    public void setSilentMode(boolean silentMode) {
+        this.silentMode = silentMode;
+    }
+
+    private CSVListener readCsvFrom(CSVListener listener, String urlSource, String fileSource){
+        try {
+            if (urlSource != null) {
+                URL url = new URL(urlSource);
+                try (InputStream is = url.openStream()) {
+                    new CSVLibrary().parse(is, listener);
+                }
+            }
+            if (fileSource != null) {
+                InputStream is = new BufferedInputStream(new FileInputStream(fileSource));
+                new CSVLibrary().parse(is, listener);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return listener;
+    }
+
+    private static class SetListener implements CSVListener {
+        private Collection<Date> inputSet = new HashSet<>();
+        SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/MM/yyyy");
+
+        @Override
+        public void handleLine(List<String> list) throws Exception {
+            inputSet.add(dateFormatter.parse(list.get(0)));
+        }
+
+        Collection<Date> returnContents(){
+            return inputSet;
+        }
+    }
+
+    private static class MapListener implements CSVListener {
+        private Map<String, Double> inputMap = new HashMap<>();
+
+        @Override
+        public void handleLine(List<String> list) {
+            inputMap.put( list.get(0),Double.parseDouble(list.get(1)));
```

Comment:
you may want to do some bounds checking on this?  Has the CSV been validated to contain two fields?